### PR TITLE
Refactor previous/next navigation to use document driven composable

### PR DESCRIPTION
**Pull request by tom-blake**
Now that we've turned on document driven mode, we can take advantage of further abstractions available through Nuxt Content's built-in composables.

More on document-driven mode here: https://content.nuxtjs.org/guide/writing/document-driven More on useContent here: https://content.nuxtjs.org/api/composables/use-document-driven

Test plan:
- Confirm that previous/next page navigation redirects to the correct page

### CODE REVIEW

File: components/Layout.vue [hunk 1]
```diff
@@ -139,14 +139,10 @@ const contentPath = path.endsWith('/') ? path.slice(0, -1) : path;
 const contentDirectory = await queryContent().where({ _path: contentPath }).findOne();
 const title = contentDirectory.title;
 const navigation = await fetchContentNavigation();
-const allLinks = navigation.flatMap((section) => section.children);
-const linkIndex = allLinks.findIndex((link) => link._path === path);
-const previousPage = allLinks[linkIndex - 1];
-const nextPage = allLinks[linkIndex + 1];
```
Comment (Ray2126) [inline components/Layout.vue:145]: So satisfying removing all this logic! Nice one 💪

File: components/Layout.vue [hunk 2]
```diff
@@ -139,14 +139,10 @@ const contentPath = path.endsWith('/') ? path.slice(0, -1) : path;
...
 const section = navigation.find((s) =>
   s.children.find((child) => child._path === path)
 );
-const { toc } = useContent();
+const { toc, prev: previousPage, next: nextPage } = useContent();
```
Comment (Ray2126) [inline components/Layout.vue:145]: Any reason why we needed to rename the variables here?

Comment (tom-blake) [reply components/Layout.vue:145]: These were the names already being used in the Vue template - that's why this diff is small 👍 

I like the idea of [including the section heading titles](https://content.nuxtjs.org/api/composables/use-document-driven) in the prev/next navigation too:

image

 I imagined these variables as `prevSectionTitle` (Component) and `nextSectionTitle` (Utils) respectively so more descriptive variables felt better here

Comment (tom-blake) [reply components/Layout.vue]: @Ray2126 I've been playing around and I don't think #3 needs to be considered anymore - I've updated the variables to use the names provided by the Nuxt composable.

Please note the new additions to the diff

**Review by Ray2126 — APPROVED** [review]

**Pull request merged**